coderd/notifications/notifications_test.go

```diff
@@ -606,7 +611,42 @@ func TestNotifierPaused(t *testing.T) {
 	}, testutil.WaitShort, testutil.IntervalFast)
 }
 
-func TestNotificationTemplatesBody(t *testing.T) {
+//go:embed events.go
+var events []byte
+
+// enumerateAllTemplates gets all the template names from the coderd/notifications/events.go file.
+// TODO(dannyk): use code-generation to create a list of all templates: https://github.com/coder/team-coconut/issues/36
+func enumerateAllTemplates(t *testing.T) ([]string, error) {
+	t.Helper()
+
+	fset := token.NewFileSet()
+
+	node, err := parser.ParseFile(fset, "", bytes.NewBuffer(events), parser.AllErrors)
+	if err != nil {
+		return nil, err
+	}
+
+	var out []string
+	// Traverse the AST and extract variable names.
+	ast.Inspect(node, func(n ast.Node) bool {
+		// Check if the node is a declaration statement.
+		if decl, ok := n.(*ast.GenDecl); ok && decl.Tok == token.VAR {
+			for _, spec := range decl.Specs {
+				// Type assert the spec to a ValueSpec.
+				if valueSpec, ok := spec.(*ast.ValueSpec); ok {
+					for _, name := range valueSpec.Names {
+						out = append(out, name.String())
+					}
+				}
+			}
+		}
```

Comment:
This is mainly personal preference, but you can reduce the indentation level with loop control:

```
if ... decl.Tok != token.VAR {
  return
}
for _, spec := range decl.Specs {
   if ... ; !ok {
      continue
   }
   out = append(out, ...)
}

```

I like to think of it as keeping the 'happy path' as much to the left as possible.

Reply:
I completely agree.
I'll admit: ChatGPT generated this snippet for me, and honestly it reads better than:

```go
	// Traverse the AST and extract variable names.
	ast.Inspect(node, func(n ast.Node) bool {
		// Check if the node is a declaration statement.
		decl, ok := n.(*ast.GenDecl)
		if !ok || decl.Tok != token.VAR {
			return true
		}
		for _, spec := range decl.Specs {
			// Type assert the spec to a ValueSpec.
			valueSpec, ok := spec.(*ast.ValueSpec)
			if !ok {
				continue
			}

			for _, name := range valueSpec.Names {
				out = append(out, name.String())
			}
		}
		return true
	})
```

WDYT?

Reply:
Discussed offline, Cian doesn't mind.